compiler/rustc_monomorphize/src/partitioning/autodiff.rs

```diff
@@ -39,6 +40,9 @@ fn adjust_activity_to_abi<'tcx>(tcx: TyCtxt<'tcx>, fn_ty: Ty<'tcx>, da: &mut Vec
                     new_activities.push(activity);
                     new_positions.push(i + 1);
                 }
+                // Now we need to figure out the size of each slice element in memory.
+                // Can we actually do that here?
```

Comment:
yes, you can just get the slice element (https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.Ty.html#method.builtin_index) and get the layout of that. During monomorphization all this information is available.